core/tauri-utils/src/platform.rs

```diff
@@ -239,9 +239,14 @@ pub fn resource_dir(package_info: &PackageInfo, env: &Env) -> crate::Result<Path
   let exe_dir = exe.parent().expect("failed to get exe directory");
   let curr_dir = exe_dir.display().to_string();
 
-  if curr_dir.ends_with(format!("{MAIN_SEPARATOR}target{MAIN_SEPARATOR}debug").as_str())
-    || curr_dir.ends_with(format!("{MAIN_SEPARATOR}target{MAIN_SEPARATOR}release").as_str())
-    || cfg!(target_os = "windows")
+  let parts: Vec<&str> = curr_dir.split(MAIN_SEPARATOR).collect();
+
+  if (cfg!(target_os = "windows") || parts.contains(&"target"))
+    && (parts[parts.len() - 1] == "debug" || parts[parts.len() - 1] == "release")
+    && (parts[parts.len() - 2] == "target"
+      || parts
+        .get(parts.len() - 3)
+        .map_or(false, |&dir| dir == "target"))
```

Comment:
I believe we can simplify this a bit and we also need to check if indexes are valid

```suggestion
  let parts: Vec<&str> = curr_dir.split(MAIN_SEPARATOR).collect();
  let len = parts.len();
  if cfg!(target_os = "windows")
    || ((len >= 2 && parts[len - 2] == "target") || (len >= 3 && parts[len - 3] == "target"))
      && (parts[len - 1] == "debug" || parts[len - 1] == "release")
```

Reply:
Thanks @amrbashir
I've test that here in my app
(Rust newbie here)